test/InitStrategy.js

```diff
@@ -0,0 +1,407 @@
+      await reset(hre.config.networks.mantle.url, Number(forkBlock));
+      const [deployer, governance, treasury, whale] = await ethers.getSigners();
+      
+      const USDC_ADDRESS = "0x09Bc4E0D864854c6aFB6eB9A9cdF58aC190D0dF9";
+      const want = await ethers.getContractAt("IERC20", USDC_ADDRESS);
+      
+      const name = "xMANTLE";
+      const symbol = "xMANTLE";
+      
+      const Vault = await ethers.getContractFactory("OnChainVault");
+      const vault = await Vault.deploy();
+      await vault.deployed();
+      await vault["initialize(address,address,address,string,string)"](
+        want.address,
+        deployer.address,
+        treasury.address,
+        name,
+        symbol
+      );
+      await vault["setDepositLimit(uint256)"](ethers.utils.parseEther("10000"));
+  
+      const InitStrategy = await ethers.getContractFactory("HopStrategy");
+      const strategy = await upgrades.deployProxy(
+        InitStrategy,
+        [vault.address, deployer.address],
+        {
+          initializer: "initialize",
+          kind: "transparent",
+          constructorArgs: [vault.address],
+          unsafeAllow: ["constructor"],
+        }
+      );
+      await strategy.deployed();
+  
+      await vault["addStrategy(address,uint256,uint256)"](
+        strategy.address,
+        10000,
+        0
+      );
+  
+      await dealTokensToAddress(whale.address, TOKENS.USDC, "1000");
+      await want
+        .connect(whale)
+        ["approve(address,uint256)"](vault.address, ethers.constants.MaxUint256);
+  
+      return {
+        vault,
+        deployer,
+        want,
+        whale,
+        governance,
+        treasury,
+        strategy,
+        want,
+      };
+    }
+  
+    async function dealTokensToAddress(
+      address,
+      dealToken,
+      amountUnscaled = "100"
+    ) {
+      const token = await ethers.getContractAt("IERC20", dealToken.address);
+  
+      await network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [dealToken.whale],
+      });
+      const tokenWhale = await ethers.getSigner(dealToken.whale);
+  
+      await network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [TOKENS.ETH.whale],
+      });
+      const ethWhale = await ethers.getSigner(TOKENS.ETH.whale);
+  
+      await ethWhale.sendTransaction({
+        to: tokenWhale.address,
+        value: utils.parseEther("50"),
+      });
+  
+      await token
+        .connect(tokenWhale)
+        .transfer(address, utils.parseUnits(amountUnscaled, dealToken.decimals));
+    }
+  
+    it("should deploy strategy", async function () {
+      const { vault, strategy } = await loadFixture(
+        deployContractAndSetVariables
+      );
+      expect(await strategy.vault()).to.equal(vault.address);
+      expect(await strategy.name()).to.equal("InitStrategy USDC");
+    });
+  
+    // it("should get reasonable prices from oracle", async function () {
```

Comment:
Why is it commented?

Reply:
This old code below requires recheck and rewrite (preferably from scratch). All that is not commented - were executed at least once.